### PR TITLE
refactor(app): update failed to connect result screens

### PR DESCRIPTION
**Pull request by koji**
# Overview
[design]
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Release%3A-Opentrons-Flex-Touchscreen-Designs?type=design&node-id=7270-116281&t=X83866LrvoYCrqIQ-0

update the failed to connect result screen.

Currently the success case will be updated by a following PR

# Test Plan
- Go to Network setup menu
- Tap Wi-Fi
- Tap a ssid / tap Join other network
- Fill out a wrong password/ssid
- Tap `Continue` to try to connect

Failed to Connect screen shows up. 
Connecting screen's update is included in https://github.com/Opentrons/opentrons/pull/12722

# Changelog
- update `FailedToConnect` component and test

# Review requests
low

# Risk assessment

### CODE REVIEW

**Comment by koji** [comment]
- [ ] check this branch on dev kit
   - [ ] success case
   - [ ] failed case incorrect password
   - [ ] failed case other errors

File: app/src/organisms/OnDeviceDisplay/SetupNetwork/FailedToConnect.tsx [hunk 1]
```diff
@@ -56,71 +55,78 @@ export function FailedToConnect({
...
+            <Flex
+              flexDirection={DIRECTION_COLUMN}
+              gridGap={SPACING.spacing4}
+              justifyContent={JUSTIFY_CENTER}
+              alignItems={ALIGN_CENTER}
+            >
+              <StyledText
+                fontSize="2rem"
+                fontWeight={TYPOGRAPHY.fontWeightSemiBold}
+                lineHeight="2.72375rem"
```
Comment (jerader) [inline app/src/organisms/OnDeviceDisplay/SetupNetwork/FailedToConnect.tsx]: i think the fontSize and lineHeight can be replaced with something in the style map. `h3`?

**Review by jerader — APPROVED** [review]
lgtm! left a comment.

File: app/src/organisms/OnDeviceDisplay/SetupNetwork/FailedToConnect.tsx [hunk 2]
```diff
@@ -56,71 +55,78 @@ export function FailedToConnect({
...
+                  ? t('incorrect_password_for_ssid', { ssid: ssid })
+                  : t('failed_to_connect_to_ssid', { ssid: ssid })}
               </StyledText>
-            )}
+              {!isInvalidPassword &&
+                requestState != null &&
+                'error' in requestState &&
+                requestState.error != null &&
+                'message' in requestState.error &&
+                requestState.error.message != null && (
```
Comment (ewagoner) [inline app/src/organisms/OnDeviceDisplay/SetupNetwork/FailedToConnect.tsx:108]: Can this whole thing be condensed down to `!isInvalidPassword && requestState?.error?.message` ?

Comment (koji) [reply app/src/organisms/OnDeviceDisplay/SetupNetwork/FailedToConnect.tsx:108]: I'm keeping this to avoid type issue.

**Review by ewagoner — APPROVED** [review]

**Pull request merged**